### PR TITLE
Root network creation

### PR DESCRIPTION
**Pull request by klesaulnier**

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/service/ConsumerService.java [hunk 1]
```diff
@@ -230,6 +242,22 @@ public Consumer<Message<String>> consumeCaseImportSucceeded() {
...
+                             Map<String, String> importParameters, UUID importReportUuid) {
+        DynamicSimulationParametersInfos dynamicSimulationParameters = DynamicSimulationService.getDefaultDynamicSimulationParameters();
+        UUID loadFlowParametersUuid = createDefaultLoadFlowParameters(userId, getUserProfile(userId));
+        UUID shortCircuitParametersUuid = createDefaultShortCircuitAnalysisParameters();
+        UUID securityAnalysisParametersUuid = createDefaultSecurityAnalysisParameters();
+        UUID sensitivityAnalysisParametersUuid = createDefaultSensitivityAnalysisParameters();
+        studyService.insertStudy(studyUuid, userId, networkInfos, caseInfos, loadFlowParametersUuid, shortCircuitParametersUuid, DynamicSimulationService.toEntity(dynamicSimulationParameters, objectMapper), null, securityAnalysisParametersUuid, sensitivityAnalysisParametersUuid, importParameters, importReportUuid);
+    }
+
+    private void recreateNetworkOfRootNetwork(StudyEntity studyEntity, UUID rootNetworkUuid, String userId, NetworkInfos networkInfos) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/ConsumerService.java:255]: recreateNetwork ?

File: src/main/java/org/gridsuite/study/server/service/ConsumerService.java [hunk 2]
```diff
@@ -198,22 +206,26 @@ public Consumer<Message<String>> consumeCaseImportSucceeded() {
...
+                        case ROOT_NETWORK_CREATION ->
+                            rootNetworkService.createRootNetworkFromRequest(studyEntity, RootNetworkInfos.builder()
+                                .id(rootNetworkUuid)
+                                .caseInfos(caseInfos)
+                                .reportUuid(importReportUuid)
+                                .networkInfos(networkInfos)
+                                .importParameters(importParameters)
+                                .build());
+                        case NETWORK_RECREATION ->
+                            recreateNetworkOfRootNetwork(studyEntity, rootNetworkUuid, userId, networkInfos);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/ConsumerService.java:228]: `updateStudyNetwork` ?

File: src/main/java/org/gridsuite/study/server/service/ConsumerService.java [hunk 3]
```diff
@@ -230,6 +242,22 @@ public Consumer<Message<String>> consumeCaseImportSucceeded() {
...
+        DynamicSimulationParametersInfos dynamicSimulationParameters = DynamicSimulationService.getDefaultDynamicSimulationParameters();
+        UUID loadFlowParametersUuid = createDefaultLoadFlowParameters(userId, getUserProfile(userId));
+        UUID shortCircuitParametersUuid = createDefaultShortCircuitAnalysisParameters();
+        UUID securityAnalysisParametersUuid = createDefaultSecurityAnalysisParameters();
+        UUID sensitivityAnalysisParametersUuid = createDefaultSensitivityAnalysisParameters();
+        studyService.insertStudy(studyUuid, userId, networkInfos, caseInfos, loadFlowParametersUuid, shortCircuitParametersUuid, DynamicSimulationService.toEntity(dynamicSimulationParameters, objectMapper), null, securityAnalysisParametersUuid, sensitivityAnalysisParametersUuid, importParameters, importReportUuid);
+    }
+
+    private void recreateNetworkOfRootNetwork(StudyEntity studyEntity, UUID rootNetworkUuid, String userId, NetworkInfos networkInfos) {
+        // TODO: what to do here ? throwing exception will provoke retried and won't notify frontend
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/ConsumerService.java:256]: `cause` ?

File: src/main/java/org/gridsuite/study/server/service/RootNetworkService.java [hunk 4]
```diff
@@ -64,10 +69,30 @@ public boolean exists(UUID rootNetworkUuid) {
         return rootNetworkRepository.existsById(rootNetworkUuid);
     }
 
+    /**
+     * Called by consumer - will create root network only if rootNetworkCreatationRequest is still in database
+     * @param studyEntity
+     * @param rootNetworkInfos
+     */
+    public void createRootNetworkFromRequest(StudyEntity studyEntity, @NonNull RootNetworkInfos rootNetworkInfos) {
```
[the remark this answers is not in the document]
Comment (klesaulnier) [reply src/main/java/org/gridsuite/study/server/service/RootNetworkService.java:77]: done

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 5]
```diff
@@ -600,9 +619,10 @@ public String getVoltageLevelSvgAndMetadata(String voltageLevelId, DiagramParame
         }
     }
 
-    private void persistentStoreWithNotificationOnError(UUID caseUuid, UUID studyUuid, String userId, UUID importReportUuid, String caseFormat, Map<String, Object> importParameters) {
+    private void persistentStoreWithNotificationOnError(UUID caseUuid, UUID studyUuid, UUID rootNetworkUuid, String userId, UUID importReportUuid, String caseFormat, Map<String, Object> importParameters, CaseImportAction caseImportAction) {
         try {
-            networkConversionService.persistentStore(caseUuid, studyUuid, userId, importReportUuid, caseFormat, importParameters);
+            //TODO: change variantId
+            networkConversionService.persistentStore(caseUuid, studyUuid, rootNetworkUuid, "first_variant_id", userId, importReportUuid, caseFormat, importParameters, caseImportAction);
```
[the remark this answers is not in the document]
Comment (klesaulnier) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java:625]: done

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 6]
```diff
@@ -279,6 +282,24 @@ public BasicStudyInfos createStudy(UUID caseUuid, String userId, UUID studyUuid,
         return basicStudyInfos;
     }
 
+    @Transactional
+    public RootNetworkCreationRequestInfos createRootNetwork(UUID studyUuid, UUID caseUuid, String caseFormat, Map<String, Object> importParameters, String userId) {
+        StudyEntity studyEntity = studyRepository.findById(studyUuid).orElseThrow(() -> new StudyException(STUDY_NOT_FOUND));
+
+        UUID importReportUuid = UUID.randomUUID();
```
[the remark this answers is not in the document]
Comment (klesaulnier) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java:289]: this is how it's done when creating study / recreating network
I agree we should investigate if it's actually useful though

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**